src/app/modules/views/sign-up/sign-up.component.ts

```diff
@@ -0,0 +1,95 @@
+import { Component, OnInit } from '@angular/core';
+import { Student, Career } from '../../interfaces/student.model';
+import { SignUpService } from '../../../services/sign-up.service';
+import { Router } from '@angular/router';
+import swal from 'sweetalert2';
+
+@Component({
+  selector: 'app-sign-up',
+  templateUrl: './sign-up.component.html',
+  styleUrls: ['./sign-up.component.css']
+})
+export class SignUpComponent implements OnInit {
+
+  public student: Student;
+  public careers: Career[];
+
+  constructor(private signUpService: SignUpService, private router: Router) { 
+    this.careers = signUpService.getCareers();
+    this.student = signUpService.newStudent();
```

Comment:
Según la [documentación de Angular](https://angular.io/tutorial/toh-pt4#call-it-in-ngoninit), no es una buena idea hacer cosas en el constructor que no sean simplemente inyectar parámetros a la clase. Considera usar `ngOnInit()` para hacer la llamada a los métodos del servicio `SignUpService`.